packages/strapi-connector-mongoose/lib/mount-models.js

```diff
@@ -291,7 +290,17 @@ module.exports = ({ models, target }, ctx) => {
           }
         }
       });
-    });
+    };
+
+    // Only sync indexes in development env while it's not possible to create complex indexes directly from models
+    // In other environments it will simply create missing indexes (those defined in the models but not present in db)
+    if (strapi.app.env === 'development') {
```

Comment:
@alexandrebodin @Convly Sometimes in the code I see `=== development` and sometimes `!== production`. Is it supposed to be the same? (ie. do we really handle other environments? I saw staging in config but not sure that we really have that case in mind when we code).
If it is supposed to be the same, can we agree on a standard? Either `=== production` && `!== production` or `===development` && `!== development`

Reply:
In this specific case, we want to restrict the use (and the side-effects) of syncIndexes to the development mode only.
By doing `=== development`, we make sure it will not impact `production` but also `test` and other environment we could have.
About your question, I think it really depends on the situation and it would be difficult to define a standard here.

Reply:
These are not the same. As you said users can create as many envs as they want. This means  `env === development` really doesn't have the same meaning as `env !== production`. Here we really only want the development env :)

Simple example si when adding a `test` env to run tests against our strapi app.